toml/src/main/java/com/fasterxml/jackson/dataformat/toml/Parser.java

```diff
@@ -0,0 +1,407 @@
+            case OFFSET_DATE_TIME:
+            case LOCAL_DATE_TIME:
+            case LOCAL_DATE:
+            case LOCAL_TIME:
+                return parseDateTime(nextState);
+            case FLOAT:
+                return parseFloat(nextState);
+            case INTEGER:
+                return parseInt(nextState);
+            case ARRAY_OPEN:
+                return parseArray(nextState);
+            case INLINE_TABLE_OPEN:
+                return parseInlineTable(nextState);
+            default:
+                throw errorContext.atPosition(lexer).unexpectedToken(firstToken, "value");
+        }
+    }
+
+    private JsonNode parseDateTime(int nextState) throws IOException {
+        String text = lexer.yytext();
+        TomlToken token = poll(nextState);
+        // the time-delim index can be [Tt ]. java.time supports only [Tt]
+        if ((token == TomlToken.LOCAL_DATE_TIME || token == TomlToken.OFFSET_DATE_TIME) && text.charAt(10) == ' ') {
+            text = text.substring(0, 10) + 'T' + text.substring(11);
+        }
+
+        if (options.parseTemporalAsJavaTime) {
+            Temporal value;
+            if (token == TomlToken.LOCAL_DATE) {
+                value = LocalDate.parse(text);
+            } else if (token == TomlToken.LOCAL_TIME) {
+                value = LocalTime.parse(text);
+            } else {
+                if (token == TomlToken.LOCAL_DATE_TIME) {
+                    value = LocalDateTime.parse(text);
+                } else if (token == TomlToken.OFFSET_DATE_TIME) {
+                    value = OffsetDateTime.parse(text);
+                } else {
+                    throw new AssertionError();
+                }
+            }
+            // todo: is pojoNode right here, or can we use rawValueNode?
+            return factory.pojoNode(value);
+        } else {
+            return factory.textNode(text);
+        }
+    }
+
+    private JsonNode parseInt(int nextState) throws IOException {
+        String text = lexer.yytext().replace("_", "");
+        pollExpected(TomlToken.INTEGER, nextState);
+        if (options.bigInteger) {
+            BigInteger v;
+            if (text.startsWith("0x") || text.startsWith("0X")) {
+                v = new BigInteger(text.substring(2), 16);
+            } else if (text.startsWith("0o") || text.startsWith("0O")) {
+                v = new BigInteger(text.substring(2), 8);
+            } else if (text.startsWith("0b") || text.startsWith("0B")) {
+                v = new BigInteger(text.substring(2), 2);
+            } else {
+                v = new BigInteger(text);
+            }
+            return factory.numberNode(v);
+        } else {
+            // "Arbitrary 64-bit signed integers (from −2^63 to 2^63−1) should be accepted and handled losslessly."
+            long v;
+            try {
+                if (text.startsWith("0x") || text.startsWith("0X")) {
+                    v = Long.parseLong(text.substring(2), 16);
```

Comment:
May want to extract this, as well as cases above, to call a helper method that takes String and base, but then also catches `NumberFormatException` and range checks that JDK throws. This because JDK exceptions are pretty unhelpful if I remember, and we want to change these into something more meaningful.

It is unfortunate that helper methods from `ParserMinimalBase` are not available (as we can't quite use that without implementing token stream, something that JsonNode provides us), like:

* `_reportOverflowLong()`
* `_reportInvalidNumber()`

etc.

These are probably easier to add when there are tests for overflow, misformatted numbers etc.